src/main/java/org/elasticsearch/index/query/TermQueryBuilder.java

```diff
@@ -147,7 +188,66 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     }
 
     @Override
+    public Query toQuery(QueryParseContext parseContext) {
+        validate();
```

Comment:
I think I would just call validate after parsing for now, which will be replaced in a later stage by the validate request infra. Makes little sense to call it from here to be honest. I would remove.